### PR TITLE
Fix extraction of tar files

### PR DESCRIPTION
**Pull request by biancarosa**
This pull request fixes the extraction of tar files. I was getting too many open files and I think we ran across the problem described here on #2 
https://blog.learngoprogramming.com/gotchas-of-defer-in-go-1-8d070894cb01

### CODE REVIEW

File: pkg/common/common.go [hunk 1]
```diff
@@ -85,11 +85,11 @@ func Untar(rc io.ReadCloser, dst string) error {
 		if err != nil {
 			return err
 		}
-		defer file.Close()
 		_, err = io.Copy(file, tarReader)
 		if err != nil {
```
Comment (boaz0) [inline pkg/common/common.go:89]: Nice catch. Just I was wondering :thinking: shouldn't we have `file.Close` under this `if` statement?
I will explain myself. Let's say I opened the file and in the middle of copying the data from `tarReader` to `file`. we don't need the file anymore and we can close it (before returning `err`). Or I am missing something here.

Comment (biancarosa) [reply pkg/common/common.go:88]: Great catch @boaz0 !

**Comment by boaz0** [comment]
CI failure is not related to @biancarosa changes.

**Review by boaz0 — APPROVED** [review]
LGTM :+1: 
Amazing job :100: @biancarosa

**Pull request merged**